src/analysis/pagerankNodeDecomposition.js

```diff
@@ -44,8 +44,11 @@ export function decompose(
         }
       ),
       (x) => -x.connectionScore,
-      // The following should be called rarely and on small objects.
-      (x) => JSON.stringify(x.connection.adjacency)
+      (x) => x.connection.adjacency.type,
+      (x) =>
+        x.connection.adjacency.type === "SYNTHETIC_LOOP"
```

Comment:
Please convert this to a `switch` with an empty-assertion. As it stands,
it is vulnerable to silently doing the wrong thing in the faces of
particular new kinds of adjacencies.